jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java

```diff
@@ -108,13 +108,17 @@ static JarType determineJarType(Path jarPath) throws IOException {
               .collect(Collectors.toList());
       List<Path> snapshotDependencies =
           allDependencies.stream().filter(isSnapshot).map(Paths::get).collect(Collectors.toList());
+      Path jarParent = jarPath.getParent() == null ? Paths.get("") : jarPath.getParent();
       if (!nonSnapshotDependencies.isEmpty()) {
         FileEntriesLayer.Builder nonSnapshotDependenciesLayerBuilder =
             FileEntriesLayer.builder().setName(DEPENDENCIES);
         nonSnapshotDependencies.forEach(
             path ->
                 nonSnapshotDependenciesLayerBuilder.addEntry(
-                    path, APP_ROOT.resolve(RelativeUnixPath.get("dependencies")).resolve(path)));
+                    jarParent.resolve(path),
```

Comment:
I just realized one thing we are missing. Remember [this output](https://github.com/GoogleContainerTools/jib/pull/2888#discussion_r520255662)? A lot of people will try Jib CLI on a standard JAR without having dependency JARs at all or in right relative locations, and they will just see `java.nio.file.NoSuchFileException: some.jar`. And they may falsely think Jib CLI is bugged by failing to load its dependency. I think it's very important to check the existence of dependency JARs locally before we delegate to jib-core, and throw an exception (`IllegalStateException`?) with a detailed message that some dependency JARs of the user app don't exist locally. Doesn't have to be in this PR, but we should not forget this.

Reply:
That is a great point! And I agree, checking this earlier + returning a meaningful error message are both super important.